state_machine/plugs/flight/lib/pycubed.py

```diff
@@ -147,289 +147,467 @@ def __init__(self):
-            sun_sensors = [sun_zn, sun_yn, sun_xn, sun_yp, sun_xp]
-            for i in sun_sensors:
-                i.enabled = False
-            self.hardware['Sun'] = True
+            sun_sensors.append(sun_xp)
         except Exception as e:
-            print('[ERROR][Sun Sensors]', e)
+            print('[ERROR][Sun Sensor +X]', e)
+            
+        # count the number of sun sensors that we have initialized
+        sun_sensor_count = 0
+        for i in sun_sensors:
+            sun_sensor_count += 1
+            i.enabled = False  # set enabled status to False
+
+        # If there is at least one sun sensor, set to True
+        if sun_sensor_count >= 1:
+            self.hardware['Sun'] = True
 
         # Initialize H-Bridges
+        coils = []
+
         try:
             drv_x = drv8830.DRV8830(self.i2c3, 0x68)  # U6
+            coils.append(drv_x)
+        except Exception as e:
+            print('[ERROR][H-Bridge U6]', e)
+            
+        try:
             drv_y = drv8830.DRV8830(self.i2c3, 0x60)  # U8
+            coils.append(drv_y)
+        except Exception as e:
+            print('[ERROR][H-Bridge U8]', e)
+            
+        try:
             drv_z = drv8830.DRV8830(self.i2c3, 0x62)  # U4
-            coils = [drv_x, drv_y, drv_z]
-            for driver in coils:
-                driver.mode = drv8830.COAST
-                driver.vout = 0
+            coils.append(drv_z)
+        except Exception as e:
+            print('[ERROR][H-Bridge U4]', e)
+            
+        coil_count = 0
+        for driver in coils:
+            driver.mode = drv8830.COAST
+            driver.vout = 0
+            coil_count += 1
+        
+        if coil_count >= 1:
             self.hardware['Coils'] = True
+
+        # Initialize burnwires
+        burnwires = []
+        try:
+            # needed to change pinout from BURN1 to PA15, as BURN1 did not support PWMOut
+            self.burnwire1 = pwmio.PWMOut(microcontroller.pin.PA15, frequency=1000, duty_cycle=0)
+            burnwires.append(self.burnwire1)
         except Exception as e:
-            print('[ERROR][H-Bridges]', e)
+            print('[ERROR][Burn Wire IC1]', e)
+        
+        try:
+            # needed to change pinout from BURN2 to PA18, as BURN2 did not support PWMOut
+            self.burnwire2 = pwmio.PWMOut(microcontroller.pin.PA18, frequency=1000, duty_cycle=0)
+            burnwires.append(self.burnwire2)
+        except Exception as e:
+            print('[ERROR][Burn Wire IC1]', e)
+
+        burnwire_count = 0
+        for burnwire in burnwires:
+            burnwire_count += 1
+        
+        if burnwire_count >= 1:
+            self.hardware['BurnWire'] = True
+        
 
+    # reinit: reinitialize radio, sd, or IMU based upon the contents of string dev
     def reinit(self, dev):
-        dev = dev.lower()
+        # dev is a whitespace-stripped string of all lowercase letters
+        dev = dev.strip().lower()
```

Comment:
This seems strange to have `reinit("               IMU")`, `reinit("imu")`, `reinit("iMu   ")`, and `reinit('IMU')` have the same behavior.

Reply:
What is the purpose of the reinit function?

Reply:
(As in where does it get called from). I'm not really sure what the use case is where having this string cleansing would be helpful.

Reply:
The only reference I remember to it is when transitioning out of low power mode, so realistically string cleansing should never be helpful from my understanding.

Reply:
Line 255 originally used dev.lower(), so I added dev.strip() to avoid more string issues (which is what I presume .lower() was called for to begin with). I weirdly enough couldn't find a place where reinit was called where passing a string would be relevant, but I kept it as is / added .strip() in the case we later choose to use reinit in a way where this would matter.

Reply:
I really don't see a world in which we accidentally pass `"                                 iMu         "` instead of `"IMU"`. Can you please give me an example in which it would result in expected behavior?